crates/astria-sequencer/src/accounts/types.rs

```diff
@@ -7,24 +9,89 @@ use serde::{
     Serialize,
 };
 
+const ADDRESS_LEN: usize = 20;
+
 /// Address represents an account address.
 #[derive(Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize, PartialEq, Eq, Debug)]
-pub(crate) struct Address(String);
+pub struct Address([u8; ADDRESS_LEN]);
 
-impl AsRef<str> for Address {
-    fn as_ref(&self) -> &str {
-        &self.0
+impl TryFrom<Vec<u8>> for Address {
+    type Error = anyhow::Error;
+
+    fn try_from(bytes: Vec<u8>) -> std::result::Result<Self, Self::Error> {
+        let arr: [u8; ADDRESS_LEN] = bytes
+            .try_into()
+            .map_err(|_| anyhow!("invalid address length"))?;
+        Ok(Address(arr))
+    }
+}
+
+impl TryFrom<&[u8]> for Address {
+    type Error = anyhow::Error;
+
+    fn try_from(bytes: &[u8]) -> std::result::Result<Self, Self::Error> {
+        let arr: [u8; ADDRESS_LEN] = bytes
+            .try_into()
+            .map_err(|_| anyhow!("invalid address length"))?;
+        Ok(Address(arr))
+    }
+}
+
+impl TryFrom<&str> for Address {
+    type Error = anyhow::Error;
+
+    fn try_from(s: &str) -> std::result::Result<Self, Self::Error> {
+        let bytes = hex::decode(s)?;
+        let arr: [u8; ADDRESS_LEN] = bytes
+            .try_into()
+            .map_err(|_| anyhow!("invalid address hex length"))?;
+        Ok(Address(arr))
+    }
+}
+
+impl TryFrom<&crate::crypto::PublicKey> for Address {
```

Comment:
in what way? would it be better to implement a `from_public_key()` for Address instead?

Reply:
Sorry, didn't see. Yeah, that would be better. But even more so: for the key it would be good to call it smth like `Address:derive_from_public_key` or something like that.